### PR TITLE
Adjust readme

### PR DESCRIPTION
**Pull request by sgnilreutr**
1. Move Abstract section into Main Goal
2. Add Table of Contents
3. Structure document with correct headings
4. Textual changes, grammar.
5. Add Saga + Thunk examples

### CODE REVIEW

**Comment by salvoravida** [comment]
Hi, @sgnilreutr thanks for the great work! I'll review it soon!

File: README.md [hunk 1]
```diff
@@ -151,12 +180,14 @@ const App = () => (
     );
 
 export default App;
+
+//...History can now be used via other components using history.push() etc.
 ```
```
Comment (salvoravida) [inline README.md:185]: What's the goal of this comment?

Comment (sgnilreutr) [reply README.md:185]: I wanted to emphasise that by implementing redux first history the behaviour of a function such as history.push or navigate() remains the same.

Comment (sgnilreutr) [reply README.md:185]: @salvoravida Is there anything else you'd like to have changed before it can be merged?

**Comment by salvoravida** [comment]
Thank you!

**Pull request merged**